src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -84,7 +81,8 @@ private DelegatedExecution getDelegatedExecutionForFieldDefinition(GraphQLFieldD
                                                                 List<MergedField> mergedFields,
                                                                 DelegatedExecution delegatedExecution,
                                                                 ExecutionStepInfo executionStepInfo) {
-        Document query = mergedFieldsToDocument.mergedSelectionSetToDocument(mergedFields);
+        Document query = SourceQueryTransformer.transform(mergedFields, context, executionStepInfo);
```

Comment:
no static calls please: they are impossible to mock. Just declare an instance variable

Reply:
I have different view on that. Why would you mock a thing that takes data and just transforms it  - in other words  SourceQueryTransformer.transform is a pure function, without any ext deps or side effects? What would be the value of testing NadelExecutionStrategy that mocks this out? Of course it is a matter of balance (what to mock and what not) but I think in case of pure functions mocking should not be used (in similar fashion you normally will not mock things like StringUtils.isBlank or others). Happy to have a discussion about that, but generally too much mocking is considered harmful by some :)
https://www.thoughtworks.com/insights/blog/mockists-are-dead-long-live-classicists

Reply:
Yes: I definitely wanna test it with mocked out dependencies. the strategy is complex enough on its own.
please have a look here for the general thinking: https://github.com/graphql-java/graphql-java/blob/master/coding-guidelines.md

Reply:
these static calls are a bad code smell and you box yourself in a corner with that: even if you are not sure when you will mock it out you still can do it.
By using an instance variable you also make clear that you have a dependency on a non trivial piece of code. A static call just hides it.

Reply:
Well, what I am trying to say is that you should never mock it cause this code depends on ourceQueryTransformer and will not work without it.  If I am doing refactor of SourceQueryTransformer that changed the behavior of it, and it was also mocked I will never see that something is wrong here. I take your point that it should not be a static method (as per the codestyle) cause it might hide the fact that it is non trivial, but disagree that it should be mocked. As long as it is pure function it should not be mocked as there are not benefits to it.